lib/kernel/doc/guides/socket_usage.md

```diff
@@ -127,6 +127,234 @@ And finally, its possible to override the global default when creating a socket
 
+
+## Example
+
+This example is intended to show how to create a simple (echo) server
+(and client).
+
+```erlang
+-module(example).
+
+-export([client/2, client/3]).
+-export([server/0, server/1, server/2]).
+
+
+%% ======================================================================
+
+%% === Client ===
+
+client(#{family := Family} = ServerSockAddr, Msg)
+  when is_list(Msg) orelse is_binary(Msg) ->
+    {ok, Sock} = socket:open(Family, stream, default),
+    ok         = socket:connect(Sock, ServerSockAddr),
+    client_exchange(Sock, Msg);
+
+client(ServerPort, Msg)
+  when is_integer(ServerPort) andalso (ServerPort > 0) ->
+    Family   = inet, % Default
+    Addr     = get_local_addr(Family), % Pick an address
+    SockAddr = #{family => Family,
+		 addr   => Addr,
+		 port   => ServerPort},
+    client(SockAddr, Msg).
+
+client(ServerPort, ServerAddr, Msg)
+  when is_integer(ServerPort) andalso (ServerPort > 0) andalso
+       is_tuple(ServerAddr) ->
+    Family   = which_family(ServerAddr),
+    SockAddr = #{family => Family,
+		 addr   => ServerAddr,
+		 port   => ServerPort},
+    client(SockAddr, Msg).
+
+%% Send the message to the (echo) server and wait for the echo to come back.
+client_exchange(Sock, Msg) when is_list(Msg) ->
+    client_exchange(Sock, list_to_binary(Msg));
+client_exchange(Sock, Msg) when is_binary(Msg) ->
+    ok = socket:send(Sock, Msg, infinity),
+    {ok, Msg} = socket:recv(Sock, byte_size(Msg), infinity),
+    ok.
+
+
+%% ======================================================================
+
+%% === Server ===
+
+server() ->
+    %% Make system choose port (and address)
+    server(0).
+
+%% This function return the port and address that it actually uses,
+%% in case server/0 or server/1 (with a port number) was used to start it.
+
+server(#{family := Family, addr := Addr, port := _} = SockAddr) ->
+    {ok, Sock} = socket:open(Family, stream, tcp),
+    ok         = socket:bind(Sock, SockAddr),
+    ok         = socket:listen(Sock),
+    {ok, #{port := Port}} = socket:sockname(Sock),
+    Acceptor = start_acceptor(Sock),
+    {ok, {Port, Addr, Acceptor}};
+
+server(Port) when is_integer(Port) ->
+    Family   = inet, % Default
+    Addr     = get_local_addr(Family), % Pick an address
+    SockAddr = #{family => Family,
+		 addr   => Addr,
+		 port   => Port},
+    server(SockAddr).
+
+server(Port, Addr)
+  when is_integer(Port) andalso (Port >= 0) andalso
+       is_tuple(Addr) ->
+    Family   = which_family(Addr),
+    SockAddr = #{family => Family,
+		 addr   => Addr,
+		 port   => Port},
+    server(SockAddr).
+
+
+%% --- Echo Server - Acceptor ---
+
+start_acceptor(LSock) ->
+    Self = self(),
+    {Pid, MRef} = spawn_monitor(fun() -> acceptor_init(Self, LSock) end),
+    receive
+	{'DOWN', MRef, process, Pid, Info} ->
+	    erlang:error({failed_starting_acceptor, Info});
+	{Pid, started} ->
+	    %% Transfer ownership
+	    socket:setopt(LSock, otp, owner, Pid),
```

Comment:
I think we should endorse `socket:setopt(LSock, {otp,owner}, Pid)` instead.
`setopt/4` is only retained for backwards compatibility.